### PR TITLE
Math Magicians: tests

### PR DESCRIPTION
**Pull request by zhorabay**
- [ ] In this branch, we:
- Set up React Testing Library;
- Wrote unit tests for the files **operate.js** and **calculate.js** using Jest.
- Created unit tests for all React components.
- Used Jest snapshots to test the components.
- Used React Testing Library to simulate user interaction.
- Ran all tests using Jest and the tests passed.

### CODE REVIEW

**Comment by zhorabay** [comment]
Commits from 'Add navbar' to 'Fix linter errors' I rebased the changes from the development branch

File: src/tests/navbar.test.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import Navbar from '../components/Navbar';
+
+test('renders Navbar component correctly', () => {
+  const { container } = render(<Navbar />);
+  expect(container).toMatchSnapshot();
```
Comment (zeff96) [inline src/tests/navbar.test.js]: - [ ]👋 Hello,

Great job on implementing unit tests for your React components! It's fantastic to see that you're taking testing seriously. However, I've noticed that some of your tests are failing. Let's take a closer look at the issues and work on fixing them together. 😊

Screenshot from 2023-09-19 18-16-43

**Review by zeff96 — CHANGES_REQUESTED** [review]
### Status: CHANGES REQUIRED ♻️ ♻️ ♻️ 
Hi @Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
close

## Highlights
:heavy_check_mark: Linters passing
:heavy_check_mark: Correct use of GitFlow

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by skyv26 — APPROVED** [review]
## Hi @zhorabay and @nilab-khwaja, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## STATUS: APPROVED 🟢 



## Highlights
- Test are *passing*  ✅ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**